apps/macro-sandbox/lib/wrappers/wrapper.py

```
@@ -0,0 +1,325 @@
+        "ArrayNth": SafeCallable(ArrayNth),
+        "TransformTrace": SafeCallable(TransformTrace),
+        "danger": SafeCallable(danger),
+        "np": SafeModule(np),
+        "pd": SafeModule(pd),
+        "scipy": SafeModule(scipy),
+        "json_module": SafeModule(json)
+    }
+    
+    try:
+        # 1s per-item timeout via SIGALRM
+        signal.signal(signal.SIGALRM, lambda signum, frame: (_ for _ in ()).throw(TimeoutError("Script execution timed out")))
```

Ruff (0.15.9)

[warning] Unused lambda argument: `signum`

(ARG005)